### PR TITLE
[SPARK-11551][DOC] Replace example code in ml-features.md using include_example

### PR DESCRIPTION
**Pull request by yinxusen**
PR on behalf of @somideshmukh, thanks!

### CODE REVIEW

**Comment by yinxusen** [comment]
ok to test

**Comment by yinxusen** [comment]
Ping @mengxr, this is for SPARK-11551. Please sign it off if looks good to you.

File: examples/src/main/scala/org/apache/spark/examples/ml/StringIndexerExample.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      .setOutputCol("categoryIndex")
+
+    val indexed = indexer.fit(df).transform(df)
+    indexed.show()
+    // $example off$
+    sc.stop()
+  }
+}
+// scalastyle:on println
+
```
Comment (BenFradet) [inline examples/src/main/scala/org/apache/spark/examples/ml/StringIndexerExample.scala]: Trailing line

File: examples/src/main/scala/org/apache/spark/examples/ml/PolynomialExpansionExample.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      .setDegree(3)
+    val polyDF = polynomialExpansion.transform(df)
+    polyDF.select("polyFeatures").take(3).foreach(println)
+    // $example off$
+    sc.stop()
+  }
+}
+// scalastyle:on println
+
+
```
Comment (BenFradet) [inline examples/src/main/scala/org/apache/spark/examples/ml/PolynomialExpansionExample.scala]: Trailing lines

File: examples/src/main/scala/org/apache/spark/examples/ml/PCAExample.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      .fit(df)
+    val pcaDF = pca.transform(df)
+    val result = pcaDF.select("pcaFeatures")
+    result.show()
+    // $example off$
+    sc.stop()
+  }
+}
+// scalastyle:on println
+
```
Comment (BenFradet) [inline examples/src/main/scala/org/apache/spark/examples/ml/PCAExample.scala]: Trailing line

File: examples/src/main/scala/org/apache/spark/examples/ml/OneHotEncoderExample.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      .setInputCol("categoryIndex")
+      .setOutputCol("categoryVec")
+    val encoded = encoder.transform(indexed)
+    encoded.select("id", "categoryVec").show()
+    // $example off$
+    sc.stop()
+  }
+}
+// scalastyle:on println
+
```
Comment (BenFradet) [inline examples/src/main/scala/org/apache/spark/examples/ml/OneHotEncoderExample.scala]: Trailing line

File: examples/src/main/scala/org/apache/spark/examples/ml/ElementWiseProductExample.scala [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      .setOutputCol("transformedVector")
+
+    // Batch transform the vectors to create new column:
+    transformer.transform(dataFrame).show()
+    // $example off$
+    sc.stop()
+  }
+}
+// scalastyle:on println
+
```
Comment (BenFradet) [inline examples/src/main/scala/org/apache/spark/examples/ml/ElementWiseProductExample.scala]: Trailing line

**Comment by BenFradet** [comment]
I notice some formatting quirks, especially for scala examples, otherwise it looks good.

However, shouldn't we take advantage of this pr to standardize the output of the examples?
For example, I think every example should end with a `show()` or `println` so the user can just c/c the example and see what it does for himself.

**Comment by yinxusen** [comment]
@BenFradet It's reasonable. I'll modify them now. Thanks for the review.

**Comment by yinxusen** [comment]
@BenFradet Does the code look good for you?

**Comment by BenFradet** [comment]
@yinxusen I'll have a look later today

**Comment by mengxr** [comment]
Merged into master and branch-1.6. Thanks!

File: docs/ml-features.md [hunk 6]
```diff
@@ -794,39 +411,7 @@ dctDf.select("featuresDCT").show(3)
...
-});
-DataFrame df = jsql.createDataFrame(data, schema);
-DCT dct = new DCT()
-  .setInputCol("features")
-  .setOutputCol("featuresDCT")
-  .setInverse(false);
-DataFrame dctDf = dct.transform(df);
-dctDf.select("featuresDCT").show(3);
-{% endhighlight %}
+{% include_example java/org/apache/spark/examples/ml/JavaDCTExample.java %}}
```
Comment (mengxr) [inline docs/ml-features.md]: Please remove the extra `}` at the end.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaBinarizerExample.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    });
+    DataFrame continuousDataFrame = jsql.createDataFrame(jrdd, schema);
+    Binarizer binarizer = new Binarizer()
+      .setInputCol("feature")
+      .setOutputCol("binarized_feature")
+      .setThreshold(0.5);
+    DataFrame binarizedDataFrame = binarizer.transform(continuousDataFrame);
+    DataFrame binarizedFeatures = binarizedDataFrame.select("binarized_feature");
+    for (Row r : binarizedFeatures.collect()) {
+    Double binarized_value = r.getDouble(0);
```
Comment (BenFradet) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaBinarizerExample.java]: indent

File: examples/src/main/python/ml/polynomial_expansion_example.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+from pyspark.mllib.linalg import Vectors
+# $example off$
+
+if __name__ == "__main__":
+    sc = SparkContext(appName="PolynomialExpansionExample")
+    sqlContext = SQLContext(sc)
+
+    # $example on$
+    df = sqlContext\
+        .createDataFrame([(Vectors.dense([-2.0, 2.3]), ),
```
Comment (BenFradet) [inline examples/src/main/python/ml/polynomial_expansion_example.py]: nit: space

**Comment by BenFradet** [comment]
LGTM, except two minor comments.

**Comment by yinxusen** [comment]
@BenFradet I'll change it in the follow-up PR https://github.com/apache/spark/pull/10193

**Pull request closed without merging**